code/modules/mob/living/carbon/combat.dm

```diff
@@ -162,6 +162,17 @@
 		if(!throwing)
 			isTackling = FALSE	//Safety from throw_at being a jerk
 		else
+			if(istype(Obstacle,/obj/machinery/door/airlock) && istype(src,/mob/living/carbon/human)) //Tackling doors while wearing an emag hacks them open
```

Comment:
How about making a new `/atom/proc/tackled` proc, instead of performing these checks here?

Reply:
Sure, that might be useful elsewhere too

Reply:
Tried this in the latest commit, let me know your thoughts